Powershell/XblConfig/PSCmdletBase.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace XblConfig
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Dynamic;
+    using System.IO;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Reflection;
+    using System.Text;
+    using Microsoft.Xbox.Services.DevTools.Authentication;
+    using Microsoft.Xbox.Services.DevTools.XblConfig;
+
+    /// <summary>
+    /// Represents the base class for all powershell commandlets to use.
+    /// </summary>
+    public abstract class PSCmdletBase : PSCmdlet, IDynamicParameters
+    {
+        /// <summary>
+        /// Gets or sets the dynamic runtime paramters.
+        /// </summary>
+        protected RuntimeDefinedParameterDictionary RuntimeParameters { get; set; }
+
+        /// <summary>
+        /// Gets or sets a boolean value indicating whether the service call requires authorization or not.
+        /// </summary>
+        protected bool RequiresAuthorization { get; set; } = true;
+
+        /// <inheritdoc />
+        protected override void BeginProcessing()
+        {
+            DevAccount account = ToolAuthentication.LoadLastSignedInUser();
+
+            if (this.RequiresAuthorization)
+            {
+                if (account == null)
+                {
+                    throw new Exception("Didn't find dev signin info, please use \"Connect-DevAccount\" to initiate.");
```

Comment:
A developer account has not been connected yet.  Please use "Connect-DevAccount" to sign-in.